aidants_connect_web/forms.py

```diff
@@ -131,3 +134,26 @@ class MandatForm(forms.Form):
         ("long", {"title": "Mandat long", "description": "(12 mois)"}),
     ]
     duree = forms.ChoiceField(choices=DUREES, required=True, initial=3)
+
+
+class RecapMandatForm(forms.Form):
+    personal_data = forms.BooleanField(
+        label=f"J’autorise mon aidant à utiliser mes données à caractère personnel."
+    )
+    brief = forms.BooleanField(label="brief")
+    otp_token = forms.CharField(
+        max_length=6, min_length=6, validators=[RegexValidator(r"^\d{6}$")]
+    )
+
+    def __init__(self, aidant, *args, **kwargs):
+        super(RecapMandatForm, self).__init__(*args, **kwargs)
+        self.aidant = aidant
+
+    def clean_otp_token(self):
```

Comment:
Cette méthode a l'air importante… et je ne vois pas où elle est appelée. Comment ça fonctionne ?

Reply:
La méthode `clean` est la méthode appelée par `Form` pour valider les champs. `clean` suivit de `_nom_du_champ` décrit la validation spécifique au champs en question.
```
The clean_<fieldname>() method is called on a form subclass – where <fieldname> is replaced with the name of the form field attribute. This method does any cleaning that is specific to that particular attribute, unrelated to the type of field that it is. This method is not passed any parameters. You will need to look up the value of the field in self.cleaned_data and remember that it will be a Python object at this point, not the original string submitted in the form (it will be in cleaned_data because the general field clean() method, above, has already cleaned the data once).

For example, if you wanted to validate that the contents of a CharField called serialnumber was unique, clean_serialnumber() would be the right place to do this. You don’t need a specific field (it’s a CharField), but you want a formfield-specific piece of validation and, possibly, cleaning/normalizing the data.

The return value of this method replaces the existing value in cleaned_data, so it must be the field’s value from cleaned_data (even if this method didn’t change it) or a new cleaned value.
```
https://docs.djangoproject.com/en/3.0/ref/forms/validation/